app/src/main/java/org/simple/clinic/summary/assignedfacility/AssignedFacilityEffectHandler.kt

```diff
@@ -2,9 +2,40 @@ package org.simple.clinic.summary.assignedfacility
 
 import com.spotify.mobius.rx2.RxMobius
 import io.reactivex.ObservableTransformer
+import org.simple.clinic.facility.Facility
+import org.simple.clinic.facility.FacilityRepository
+import org.simple.clinic.patient.PatientRepository
+import org.simple.clinic.util.Optional
+import org.simple.clinic.util.extractIfPresent
+import org.simple.clinic.util.scheduler.SchedulersProvider
+import java.util.UUID
+import java.util.function.Function
+
+class AssignedFacilityEffectHandler(
+    private val patientRepository: PatientRepository,
+    private val facilityRepository: FacilityRepository,
+    private val schedulersProvider: SchedulersProvider
+) {
 
-class AssignedFacilityEffectHandler {
   fun build(): ObservableTransformer<AssignedFacilityEffect, AssignedFacilityEvent> = RxMobius
       .subtypeEffectHandler<AssignedFacilityEffect, AssignedFacilityEvent>()
+      .addTransformer(LoadAssignedFacility::class.java, loadAssignedFacility())
       .build()
+
+  private fun loadAssignedFacility(): ObservableTransformer<LoadAssignedFacility, AssignedFacilityEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .flatMap { patientRepository.patient(it.patientUuid) }
```

Comment:
Yes, Initially I made it reactive because I thought I have to update the name when facility changes. But since we are getting `Facility` from `FacilitySelectionActivity` we no longer need it.